### PR TITLE
Passcode Module Update

### PR DESCRIPTION
**Pull request by umerkhan-apple**
- Updating ORKTest to allow for more visible passcode creation and removal items.
- Updated error handling to be more explicit.
- Added additional documentation to ORKPasscodeViewController.

Issue #485

### CODE REVIEW

**Comment by umerkhan-apple** [comment]
@YuanZhu-apple Please review. :)

File: ResearchKit/Common/ORKPasscodeStepViewController.m [hunk 1]
```diff
@@ -417,6 +420,14 @@ - (void)wrongAttempt {
     });
 }
 
+- (void)throwPasscodeNotFoundExceptionWithError:(NSError *)error {
```
Comment (YuanZhu-apple) [inline ResearchKit/Common/ORKPasscodeStepViewController.m]: How about name it `throwExceptionWithKeychainError:`?

File: ResearchKit/Common/ORKPasscodeViewController.h [hunk 2]
```diff
@@ -80,12 +80,46 @@ ORK_CLASS_AVAILABLE
 
 - (instancetype)init NS_UNAVAILABLE;
 
+/**
+ An authenticaiton passcode view controller allows a user to be authenticated using
+ the passcode stored in the keychain.
+ 
+ You must check to see if the passcode is stored in the keychain before presenting
```
Comment (YuanZhu-apple) [inline ResearchKit/Common/ORKPasscodeViewController.h]: Can you mention `isPasscodeStoredInKeychain` in this doc? And refers passcodeStep for creation as well?

File: ResearchKit/Common/ORKPasscodeStepViewController.m (hunk 1 above)
Comment (umerkhan-apple) [inline ResearchKit/Common/ORKPasscodeStepViewController.m]: Oh yes, I like that!

**Comment by umerkhan-apple** [comment]
@YuanZhu-apple updated!

**Comment by YuanZhu-apple** [comment]
r=me.

**Pull request merged**